orangecontrib/text/widgets/owpubmed.py

```diff
@@ -0,0 +1,463 @@
+                h_box, self, '',
+                callback=lambda: self.open_calendar(self.pub_date_to_input),
+                tooltip='Pick a date using the calendar widget.')
+        open_calendar_button_to.setMaximumSize(open_calendar_button_from
+                                               .sizeHint())
+        open_calendar_button_to.setIcon(QIcon(_i('calendar.svg')))
+        open_calendar_button_to.setIconSize(QtCore.QSize(16, 16))
+        open_calendar_button_to.setFocusPolicy(QtCore.Qt.NoFocus)
+        open_calendar_button_to.setSizePolicy(QtGui.QSizePolicy.Fixed,
+                                              QtGui.QSizePolicy.Fixed)
+
+        self.pubmed_controls.append(self.pub_date_from_input)
+        self.pubmed_controls.append(self.pub_date_to_input)
+
+        # Keywords.
+        h_box = gui.widgetBox(regular_search_box, orientation='horizontal')
+        label = gui.label(h_box, self, 'Query:')
+        label.setMaximumSize(label.sizeHint())
+        self.keyword_combo = QComboBox(h_box)
+        self.keyword_combo.setMinimumWidth(150)
+        self.keyword_combo.setEditable(True)
+        h_box.layout().addWidget(self.keyword_combo)
+        self.keyword_combo.activated[int].connect(self.select_keywords)
+        self.pubmed_controls.append(self.keyword_combo)
+
+        # --- Advanced search ---
+        advanced_search_box = gui.widgetBox(self.controlArea, addSpace=True)
+        # Advanced search query.
+        h_box = gui.widgetBox(advanced_search_box, orientation='horizontal')
+        self.advanced_query_input = QTextEdit(h_box)
+        h_box.layout().addWidget(self.advanced_query_input)
+        self.pubmed_controls.append(self.advanced_query_input)
+
+        gui.createTabPage(self.search_tabs, 'Regular search',
+                          regular_search_box)
+        gui.createTabPage(self.search_tabs, 'Advanced search',
+                          advanced_search_box)
+
+        # Search info label.
+        self.search_info_label = gui.label(
+                self.controlArea, self,
+                'Number of retrievable records for this search query: /')
+
+        # Search for records button.
+        self.run_search_button = gui.button(
+                self.controlArea,
+                self,
+                'Find records',
+                callback=self.run_search,
+                tooltip='Performs a search for articles that fit the '
+                        'specified parameters.')
+        self.pubmed_controls.append(self.run_search_button)
+
+        h_line = QFrame()
+        h_line.setFrameShape(QFrame.HLine)
+        h_line.setFrameShadow(QFrame.Sunken)
+        self.controlArea.layout().addWidget(h_line)
+
+        # RECORD RETRIEVAL
+        # Text includes box.
+        text_includes_box = gui.widgetBox(self.controlArea,
+                                          'Text includes', addSpace=True)
+        self.authors_chbox = gui.checkBox(text_includes_box, self,
```

Comment:
`authors_checkbox` is just 3 characters longer and much easier to read. Same holds for title, mesh and abstract.